File: app/src/main/java/com/ericneidhardt/dynamicsoundboard/dialog/BaseDialog.java
package com.ericneidhardt.dynamicsoundboard.dialog;

import android.app.DialogFragment;
import com.ericneidhardt.dynamicsoundboard.soundmanagement.ServiceManagerFragment;
import com.ericneidhardt.dynamicsoundboard.soundsheet.SoundSheetManagerFragment;

/**
 * File created by eric.neidhardt on 14.11.2014.
 */
abstract class BaseDialog extends DialogFragment
{
	ServiceManagerFragment getServiceManagerFragment()
	{
		return (ServiceManagerFragment)this.getFragmentManager().findFragmentByTag(ServiceManagerFragment.TAG);
	}

	SoundSheetManagerFragment getSoundSheetManagerFragment()
	{
		return (SoundSheetManagerFragment)this.getFragmentManager().findFragmentByTag(SoundSheetManagerFragment.TAG);
	}
}

File: app/src/main/kotlin/org/neidhardt/dynamicsoundboard/soundactivity/BaseDialog.kt
package org.neidhardt.dynamicsoundboard.soundactivity

import android.support.v4.app.DialogFragment
import org.neidhardt.dynamicsoundboard.soundactivity.SoundActivity

/**
 * File created by eric.neidhardt on 14.11.2014.
 */
abstract class BaseDialog : DialogFragment()
{
	protected val KEY_CALLING_FRAGMENT_TAG: String = "KEY_CALLING_FRAGMENT_TAG"

	val soundActivity: SoundActivity
		get() = this.activity as SoundActivity
}

